[Bug] String cannot contain newlines

```
Error executing vim.schedule lua callback: ...e/nvim/site/pack/packer/start/fidget.nvim/lua/fidget.lua:218: String
 cannot contain newlines
stack traceback:
        [C]: in function 'nvim_buf_set_lines'
        ...e/nvim/site/pack/packer/start/fidget.nvim/lua/fidget.lua:218: in function 'show'
        ...e/nvim/site/pack/packer/start/fidget.nvim/lua/fidget.lua:55: in function 'render_fidgets'
        ...e/nvim/site/pack/packer/start/fidget.nvim/lua/fidget.lua:169: in function 'fmt'
        ...e/nvim/site/pack/packer/start/fidget.nvim/lua/fidget.lua:387: in function <...e/nvim/site/pack/packer/s
tart/fidget.nvim/lua/fidget.lua:386>
```

Happens when running go test files using codelens

1. Open go test file
2. Move cursor to a test function
3. `:lua vim.lsp.codelens.run()`
4. Error (but the test run fine ofc)

I encountered the same error when using Go's Test method, and I captured some logs.
### Screenshot
<img width="1200" alt="image" src="https://user-images.githubusercontent.com/8832748/164998924-4a76defc-712b-44e1-bdbb-38fb91e5c408.png">

### Code
```go
package main

import (
	"testing"
)

func TestFidget(t *testing.T) {
	t.Error("error")
}
```
Run when the cursor is on the `TestFidget` line: `:lua vim.lsp.codelens.run()`
### log
```Log
[DEBUG 一  4/25 06:09:24 2022] ...e/nvim/site/pack/packer/start/fidget.nvim/lua/fidget.lua:342: Received progress notification: {
  info = {
    client_id = 1,
    method = "$/progress"
  },
  msg = {
    token = "6129484611666145821",
    value = {
      cancellable = true,
      kind = "begin",
      message = "Running...",
      title = "Running go test"
    }
  }
}
[DEBUG 一  4/25 06:09:25 2022] ...e/nvim/site/pack/packer/start/fidget.nvim/lua/fidget.lua:342: Received progress notification: {
  info = {
    client_id = 1,
    method = "$/progress"
  },
  msg = {
    token = "6129484611666145821",
    value = {
      cancellable = true,
      kind = "report",
      message = "=== RUN   TestFidget\n    main_test.go:13: error\n--- FAIL: TestFidget (0.00s)\nFAIL"
    }
  }
}
[DEBUG 一  4/25 06:09:26 2022] ...e/nvim/site/pack/packer/start/fidget.nvim/lua/fidget.lua:342: Received progress notification: {
  info = {
    client_id = 1,
    method = "$/progress"
  },
  msg = {
    token = "6129484611666145821",
    value = {
      cancellable = true,
      kind = "report",
      message = "FAIL\tdoudian\t0.320s"
    }
  }
}
[DEBUG 一  4/25 06:09:26 2022] ...e/nvim/site/pack/packer/start/fidget.nvim/lua/fidget.lua:342: Received progress notification: {
  info = {
    client_id = 1,
    method = "$/progress"
  },
  msg = {
    token = "6129484611666145821",
    value = {
      cancellable = true,
      kind = "report",
      message = "FAIL"
    }
  }
}
```

--- a/lua/fidget.lua
+++ b/lua/fidget.lua
@@ -193,6 +193,20 @@ function base_fidget:fmt()
   render_fidgets()
 end
 
+local function splits_on_newlines(lines)
+  local result = {}
+  for _, line in ipairs(lines) do
+    if line:match("\n") then
+      for _, subline in ipairs(vim.split(line, "\n")) do
+        table.insert(result, subline)
+      end
+    else
+      table.insert(result, line)
+    end
+  end
+  return result
+end
+
 function base_fidget:show(offset)
   local height = #self.lines
   local width = self.max_line_len
@@ -239,6 +253,7 @@ function base_fidget:show(offset)
 
   api.nvim_win_set_option(self.winid, "winblend", options.window.blend)
   api.nvim_win_set_option(self.winid, "winhighlight", "Normal:FidgetTask")
+  self.lines = splits_on_newlines(self.lines) -- handle lines that might contain a "\n" character
   api.nvim_buf_set_lines(self.bufid, 0, height, false, self.lines)
   if options.fmt.stack_upwards then
     api.nvim_buf_add_highlight(self.bufid, -1, "FidgetTitle", height - 1, 0, -1)

